src/iexamine.cpp

```diff
@@ -3598,7 +3598,7 @@ void iexamine::autodoc( player &p, const tripoint &examp )
                 popup_player_or_npc( patient, _( "You have already installed this bionic." ),
                                      _( "%1$s has already installed this bionic." ) );
                 return;
-            } else if( bid->upgraded_bionic && !patient.has_bionic( bid->upgraded_bionic ) ) {
+            } else if( bid->upgraded_bionic && !patient.has_bionic( bid->upgraded_bionic ) && it->is_upgrade() ) {
```

Comment:
This is more a property of the bionic, not of the item that is used to install it.

Reply:
Both bio_advreactor and bio_reactor_upgrade refer to the bionic called bio_advreactor, so you have to check the item if it is an upgrade of an existing bionic or not.